redirect/redirect.c

```diff
@@ -0,0 +1,94 @@
+/*
+* SPDX-FileCopyrightText: (c) 2024 Ring Zero Desenvolvimento de Software LTDA
+* SPDX-License-Identifier: GPL-2.0-only
+*/
+
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_endian.h>
+
+#define ETH_ALEN 6
+
+#define memcmp __builtin_memcmp
+#define memcpy __builtin_memcpy
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __uint(max_entries, 32);
+    __type(key, char[8]);
+    __type(value, u8[ETH_ALEN]);
+} macaddr_map SEC(".maps");
+
+struct {
+    __uint(type, BPF_MAP_TYPE_DEVMAP);
+    __uint(max_entries, 8);
+    __type(key, int);
+    __type(value, int);
+} ifindex_map SEC(".maps");
+
+typedef struct redirect_config_s {
+	struct {
+		u8 *gateway1;
+		u8 *gateway2;
+		u8 *client;
+		u8 *dome;
+	} macaddr;
+} redirect_config_t;
+
+#define REDIRECT_CONFIG_CHECK(attr, map, key) do {		\
+	(attr) = bpf_map_lookup_elem((map), (key));		\
+	if ((attr) == NULL) {					\
+		bpf_printk("missing configuration: %s", key);	\
+		return false;					\
+	}							\
+} while(0)
+
+static bool redirect_config(redirect_config_t *config)
+{
+	REDIRECT_CONFIG_CHECK(config->macaddr.gateway1, &macaddr_map, "gateway1");
+	REDIRECT_CONFIG_CHECK(config->macaddr.gateway2, &macaddr_map, "gateway2");
+	REDIRECT_CONFIG_CHECK(config->macaddr.client,   &macaddr_map, "client  ");
+	REDIRECT_CONFIG_CHECK(config->macaddr.dome,     &macaddr_map, "dome    ");
```

Comment:
because the `bpf_map` requires the field to be of a fixed size, and I couldn't manage to make it work as a null terminated string. I tried hard, even passing zeroes instead, but if I fill the data with zeroes and pass the null terminated string to the API, it fails. maybe we could dig this further later on?